### PR TITLE
999 generic http connector

### PR DESCRIPTION
**Pull request by jonahx**
TBD but....

Quick explanation:

- Creates a generic http connector
- Integration test shows how to use it to recreated Basic Auth connector
- Includes unit tests, though they aren't exhaustive.

### CODE REVIEW

File: internal/plugin/connectors/http/generic/connector.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"fmt"
+	gohttp "net/http"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Connector injects an HTTP request with AWS authorization headers.
```
Comment (BradleyBoutcher) [inline internal/plugin/connectors/http/generic/connector.go]: Should this say AWS?

File: http-generic.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: 2
```
[the remark this answers is not in the document]
Comment (jonahx) [reply http-generic.yml]: Good catch.  That's an error.  Should have been deleted.

File: internal/plugin/connectors/http/generic/config_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package generic
```
[the remark this answers is not in the document]
Comment (jonahx) [reply internal/plugin/connectors/http/generic/config_test.go]: No, in go white box unit test files (which is what this is), and often even black box unit tests as well, go in the same directory as the files they're testing.  Here's a random example from the stdlib:

https://golang.org/src/bytes/

File: pkg/secretless/plugin/sharedobj/internal_plugins.go [hunk 4]
```diff
@@ -20,9 +21,10 @@ func GetInternalPluginsFunc() (plugin.AvailablePlugins, error) {
 	// New connectors should have an entry in the map below, according to their type (HTTP/TCP)
 	return &Plugins{
 		HTTPPluginsByID: map[string]http.Plugin{
-			"aws":        aws.GetHTTPPlugin(),
-			"basic_auth": basicauth.GetHTTPPlugin(),
-			"conjur":     conjur.GetHTTPPlugin(),
+			"aws":          aws.GetHTTPPlugin(),
+			"basic_auth":   basicauth.GetHTTPPlugin(),
+			"generic_http": generic.GetHTTPPlugin(),
+			"conjur":       conjur.GetHTTPPlugin(),
```
Comment (doodlesbykumbi) [inline pkg/secretless/plugin/sharedobj/internal_plugins.go]: nit: sort keys

File: test/http_generic/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      interval: 1s
+      timeout: 30s
+    volumes:
+      - ./http-basic-auth.conf:/etc/nginx/conf.d/default.conf:ro
+      - ./basic_auth.htpasswd:/etc/nginx/.htpasswd:ro
+
+  secretless:
+    build:
+      context: ../..
+    environment:
```
Comment (doodlesbykumbi) [inline test/http_generic/docker-compose.yml]: These envvars aren't use anywhere as far as I can tell. If not let's get rid of them

Comment (jonahx) [reply test/http_generic/docker-compose.yml]: I'll try it and see we'll see if anything breaks :)

File: test/mssql_connector/start [hunk 6]
```diff
@@ -13,7 +13,7 @@ done
 # the order of the services is important. mssql must be up before we start secretless
 docker-compose up -d mssql
 
-./wait_for_mssql
+time ./wait_for_mssql
```
Comment (doodlesbykumbi) [inline test/mssql_connector/start]: hm ?

**Pull request merged**